src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -0,0 +1,352 @@
+
+    def process(self):
+        """
+        Intended to be the main method which can perform all relevant processing, and/or write to a target file, which depends on the type of processor.
+        For SnowparkAnnotationProcessor, the target file is the setup script.
+        """
+        # 1. Get all src py files
+        all_py_files_to_process = self.get_all_py_files_to_process()
+
+        # 2. Get all src.py -> dest.py mapping
+        src_py_file_to_dest_py_file_map = self.get_src_py_file_to_dest_py_file_map()
+
+        # 3. Get entities through Snowpark callback
+        src_py_file_to_collected_entities: Dict[Path, ExtensionFunctionProperties] = {}
+        for py_file in all_py_files_to_process:
+            try:
+                collected_entities = execute_in_sandbox(
+                    py_file=py_file.resolve(), kwargs=self.kwargs
+                )
+            except ClickException:
+                cc.warning(
+                    f"Error processing extension functions in {py_file}, skipping generating code of all objects from this file."
+                )
+                collected_entities = []
+            src_py_file_to_collected_entities[py_file] = collected_entities
+
+            # 4. Enrich entities by setting additional properties
+            for entity in collected_entities:
+                entity.set_additional_properties(
+                    py_file=py_file, dest_file=src_py_file_to_dest_py_file_map[py_file]
+                )
+
+    def get_all_py_files_to_process(self) -> List[Path]:
+        """
+        For the project definition for a native app, find all python files in source that need to be consumed by the Snowpark Annotation Processor.
+        """
+
+        src_py_files: List[Path] = []
+        artifact_src = self.artifact_to_process.src
+
+        # Case 1: When artifact has the following src/dest pairing
+        # src: john/doe/folder1/*.py OR john/doe/folder1/**/*.py
+        # dest: stagepath/
+        if is_glob(artifact_src) and artifact_src.endswith(".py"):
+            src_py_files.extend(list(self.project_root.glob(artifact_src)))
+
+        # Case 2: When artifact has the following src/dest pairing
+        # src: john/doe/folder1/**/* (in this case, all files and directories under src need to be considered)
+        # dest: stagepath/
+        elif "**" in artifact_src:
+            all_source_paths = list(self.project_root.glob(artifact_src))
+            src_py_files.extend(get_all_python_files(all_source_paths))
+
+        # Case 3: When artifact has the following src/dest pairing
+        # src: john/doe/folder1/*
+        # dest: stagepath/ (in this case, the directories under folder1 will be symlinked, which means files inside those directories also need to be considered due to implicit availability from directory symlink)
+        elif is_glob(artifact_src):
```

Comment:
Guy added a way to keep track of the mapping during bundle, we should just use that. There is no reason to do resolve globbing here again. It's bad for separation of concerns and maintainability.

Reply:
Synced with @sfc-gh-gbloom . Seems like neither the mapping nor `source_path_to_deploy_path` will address the use here. The mapping can be "incomplete" in that it doesn't contain symlinked kv pairs. 

Also, for `source_path_to_deploy_path`, I'd need access to
1. each source (file) path for a given dir, and pass it to `source_path_to_deploy_path`, i.e. I need enumeration anyway for all source paths
2. the exact file location in the deploy root, not just a parent path that could be returned from `source_path_to_deploy_path`.